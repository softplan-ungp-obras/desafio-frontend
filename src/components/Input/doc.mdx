---
name: Input
menu: Components
---

import { Playground, PropsTable } from 'docz'
import Input from './'

# Input

<PropsTable of={Input} />

## Basic usage

<Playground>
  <Input
    label="Input text"
    onSubmit={() => null}
    placeholder="simple imput"
    refInput="this.inputRef"
    required
  />
  <Input
    onSubmit={() => null}
    placeholder="search"
    refInput="this.inputRef"
    required
    search
  />
  <Input
    clean
    label="Clean Input text"
    onSubmit={() => null}
    placeholder="simple imput"
    refInput="this.inputRef"
    required
  />
</Playground>
