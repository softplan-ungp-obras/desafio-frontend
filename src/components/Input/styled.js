import styled, { css } from 'styled-components'

export const GroupStyled = styled.div`
  position: relative;
  margin: 10px 0 25px;
  display: flex;
  flex-direction: column;
  align-items: flex-start;

  ${props => props.search && css`
    flex-direction: row;
    align-items: center;
  `}

  input {
    padding: ${props => props.padding};
    padding-right: 30px;
    width: ${props => props.width};
    background: #fff;
    border: 1px solid #b6b9d0;
    box-sizing: border-box;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.18);
    font-size: 1rem;
    color: #696d8c;
    outline: none;
    transition: all .2s linear;

    &:focus,
    &:hover {
      outline: none;
      box-shadow: 0 15px 30px -15px rgba(164, 164, 164);
    }

    ${props => props.clean && css`
      border: 0;
      border-bottom: 2px solid #9e9e9e;
      box-shadow: none;
      padding: ${({ padding }) => padding} ;
      width: ${({ width }) => width};

      &:focus,
      &:hover {
        border-bottom: 2px solid #005b95;
        box-shadow: none;
        outline: none;
      }
    `}
  }

  .icon {
    width: 20px;
    height: 20px;
    position: absolute;
    right: 10px;
    background: none;
    border: 0;
    padding: 0;
    margin: 0;
    cursor: pointer;
  }
`

export const Label = styled.label`
  color: #696d8c;
  font-size: 1.2em;
  margin-bottom: 5px;
`

export const InputStyled = styled.input`
  margin: 10px 0 25px;
  padding: 10px;
  width: 100%;
  background: #fff;
  border: 1px solid #b6b9d0;
  box-sizing: border-box;
  box-shadow: inset 0px 3px 3px rgba(0, 0, 0, 0.05);
  font-size: 1rem;
  color: #696d8c;
  outline: none;
  transition: all .2s linear;

  &:focus {
    outline: none;
  }
`
