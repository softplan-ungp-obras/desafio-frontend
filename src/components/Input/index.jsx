import React from 'react'
import PropTypes from 'prop-types'
import { ReactComponent as SearchIcon } from './search.svg'
import { GroupStyled, Label } from './styled'


const Input = (props) => {
  const {
    type,
    search,
    refInput,
    focus,
    className,
    label,
    onSubmit,
    padding,
    placeholder,
    required,
    width,
  } = props

  return (
    <GroupStyled {...props}>
      {label && <Label>{label}</Label>}
      <input
        ref={refInput}
        autoFocus={focus}
        className={className}
        padding={padding}
        placeholder={placeholder}
        required={required}
        type={type}
        width={width}
      />
      {search && (
        <button className="icon" onClick={event => onSubmit(event)} type="button">
          <SearchIcon />
        </button>
      )}
    </GroupStyled>
  )
}

Input.defaultProps = {
  className: '',
  focus: false,
  label: '',
  onSubmit: null,
  padding: '10px',
  placeholder: '',
  required: false,
  search: false,
  type: 'text',
  width: '100%',
}

Input.propTypes = {
  className: PropTypes.string,
  focus: PropTypes.bool,
  label: PropTypes.string,
  onSubmit: PropTypes.func,
  padding: PropTypes.string,
  placeholder: PropTypes.string,
  refInput: PropTypes.instanceOf(Object).isRequired,
  required: PropTypes.bool,
  search: PropTypes.bool,
  type: PropTypes.string,
  width: PropTypes.string,
}

export default Input
