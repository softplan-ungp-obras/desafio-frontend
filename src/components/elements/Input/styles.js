import styled, { css } from 'styled-components';
import theme from '~/styles/theme';

export const Box = styled.div`
  position: relative;
  input {
    border: 1px solid ${theme.lighGrey.main};
    padding: 15px;
    height: 45px;
    width: 100%;
    box-shadow: 0px 3px 2px ${theme.lighGrey.main};
    color: ${theme.grey.main};
    font-size: 15px;
    font-weight: 600;
  }

  ${props =>
    props.level === 'normal' &&
    css`
      input {
        height: 35px;
        box-shadow: none;
        border-color: transparent;
        border-bottom: 2px solid ${theme.grey.main};
      }
    `}
`;
export const Icon = styled.span`
  position: absolute;
  top: 55%;
  transform: translate(0, -50%);
  right: 10px;
  font-size: 24px;
  color: ${theme.grey.main};
  ${props =>
    props.iconPointer &&
    css`
      cursor: pointer;
    `}
    background:  ${theme.white.main};
`;
