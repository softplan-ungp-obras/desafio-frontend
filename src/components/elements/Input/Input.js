import React from 'react';
import PropTypes from 'prop-types';
import { MdSearch } from 'react-icons/md';
import { Box, Icon } from './styles';
import { Message } from '~/components/elements';

export default function Input({
  onChange,
  onSubmit,
  icon,
  type,
  label,
  htmlFor,
  name,
  value,
  placeholder,
  onBlur,
  disabled,
  className,
  iconPointer,
  level,
  errors,
}) {
  function iconCondition() {
    switch (icon) {
      case 'SEARCH':
        return <MdSearch />;
      default:
        return <MdSearch />;
    }
  }

  return (
    <Box level={level} className={className} disabled={disabled}>
      <label htmlFor={htmlFor}>{label}</label>

      <input
        id={htmlFor}
        defaultValue={value}
        onChange={onChange}
        onBlur={onBlur}
        type={type}
        name={name}
        placeholder={placeholder}
        disabled={disabled}
      />
      {icon && (
        <Icon iconPointer={iconPointer} onClick={onSubmit}>
          {iconCondition()}
        </Icon>
      )}
      {errors && <Message type="error">{errors}</Message>}
    </Box>
  );
}

Input.defaultProps = {
  type: 'text',
  level: 'large',
  placeholder: '',
  icon: '',
  htmlFor: '',
  className: '',
  label: '',
  onBlur: () => {},
  onSubmit: () => {},
  disabled: false,
  iconPointer: false,
  value: '',
  errors: '',
};

Input.propTypes = {
  level: PropTypes.oneOf(['large', 'normal']),
  htmlFor: PropTypes.string,
  iconPointer: PropTypes.bool,
  className: PropTypes.string,
  icon: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func,
  onBlur: PropTypes.func,
  type: PropTypes.string,
  name: PropTypes.string.isRequired,
  label: PropTypes.string,
  value: PropTypes.string,
  placeholder: PropTypes.string,
  disabled: PropTypes.bool,
  errors: PropTypes.string,
};
