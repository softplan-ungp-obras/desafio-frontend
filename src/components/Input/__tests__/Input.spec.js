import React from 'react'
import { shallow, mount } from 'enzyme'
import 'jest-styled-components'
import Input from '..'
import 'setupTest'

describe('Input.', () => {
  it('Should be render <Input />.', () => {
    const inputRef = React.createRef()
    const onChange = jest.fn()
    const component = shallow(
      <Input
        refInput={inputRef}
        className="class"
        label="E-mail"
        onChange={onChange}
        required
      />
    )
    expect(component).toMatchSnapshot()
  })
})
