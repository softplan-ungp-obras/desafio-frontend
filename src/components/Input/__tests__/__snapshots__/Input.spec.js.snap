// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input. Should be render <Input />. 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    className="class"
    focus={false}
    label="E-mail"
    onChange={[MockFunction]}
    onSubmit={null}
    padding="10px"
    placeholder=""
    refInput={
      Object {
        "current": null,
      }
    }
    required={true}
    search={false}
    type="text"
    width="100%"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.label>
          E-mail
        </styled.label>,
        <input
          autoFocus={false}
          className="class"
          padding="10px"
          placeholder=""
          required={true}
          type="text"
          width="100%"
        />,
        false,
      ],
      "className": "class",
      "focus": false,
      "label": "E-mail",
      "onChange": [MockFunction],
      "onSubmit": null,
      "padding": "10px",
      "placeholder": "",
      "refInput": Object {
        "current": null,
      },
      "required": true,
      "search": false,
      "type": "text",
      "width": "100%",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "E-mail",
        },
        "ref": null,
        "rendered": "E-mail",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "autoFocus": false,
          "className": "class",
          "padding": "10px",
          "placeholder": "",
          "required": true,
          "type": "text",
          "width": "100%",
        },
        "ref": Object {
          "current": null,
        },
        "rendered": null,
        "type": "input",
      },
      false,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.label>
            E-mail
          </styled.label>,
          <input
            autoFocus={false}
            className="class"
            padding="10px"
            placeholder=""
            required={true}
            type="text"
            width="100%"
          />,
          false,
        ],
        "className": "class",
        "focus": false,
        "label": "E-mail",
        "onChange": [MockFunction],
        "onSubmit": null,
        "padding": "10px",
        "placeholder": "",
        "refInput": Object {
          "current": null,
        },
        "required": true,
        "search": false,
        "type": "text",
        "width": "100%",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "E-mail",
          },
          "ref": null,
          "rendered": "E-mail",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoFocus": false,
            "className": "class",
            "padding": "10px",
            "placeholder": "",
            "required": true,
            "type": "text",
            "width": "100%",
          },
          "ref": Object {
            "current": null,
          },
          "rendered": null,
          "type": "input",
        },
        false,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
